dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerClusters.java

```diff
@@ -149,12 +169,12 @@ public void onServerAdded(WorkerServerMetadata workerServer) {
     @Override
     public void onServerRemove(WorkerServerMetadata workerServer) {
         workerMapping.remove(workerServer.getAddress(), workerServer);
-        synchronized (workerGroupMapping) {
-            List<String> removeWorkerGroupAddrList = workerGroupMapping.get(workerServer.getWorkerGroup());
+        synchronized (configWorkerGroupMapping) {
+            List<String> removeWorkerGroupAddrList = configWorkerGroupMapping.get(workerServer.getWorkerGroup());
```

Comment:
I just find the worker group at `WorkerServerMetadata` can be null, we need to make sure this value not null when worker submit heartbeat, if the value is null, we need to set to default at worker side.

Reply:
Fixed.